code/game/objects/effects/decals/cleanable/chempiles.dm

```diff
@@ -0,0 +1,79 @@
+///HIPPIESTATION MARCHES ON.
+
+GLOBAL_LIST_EMPTY(chempiles)
+/obj/effect/decal/cleanable/chempile
+	name = "chemicals"
+	desc = "An indiscernible mixture of chemicals"
+	icon = 'icons/effects/32x32.dmi'
+	icon_state = "chempile"
+	mergeable_decal = FALSE
+
+/obj/effect/decal/cleanable/chempile/examine(mob/user)
+	. = ..()
+	to_chat(user, "It contains:")//this is so fucking funny I don't want to fix it
+	if(reagents.reagent_list.len)
+		if(user.can_see_reagents()) //Show each individual reagent
+			for(var/datum/reagent/R in reagents.reagent_list)
+				to_chat(user, "[R.volume] units of [R.name]")
+		else //Otherwise, just show the total volume
+			var/total_volume = 0
+			for(var/datum/reagent/R in reagents.reagent_list)
+				total_volume += R.volume
+			to_chat(user, "[total_volume] units of various reagents")
+	else
+		to_chat(user, "Nothing.")
```

Comment:
```suggestion
	. += "<span class='notice'>It contains:</span>" //this is so fucking funny I don't want to fix it
	if(reagents.reagent_list.len)
		if(user.can_see_reagents()) //Show each individual reagent
			for(var/datum/reagent/R in reagents.reagent_list)
				. += "<span class='notice'>[R.volume] units of [R.name]</span>"
		else //Otherwise, just show the total volume
			var/total_volume = 0
			for(var/datum/reagent/R in reagents.reagent_list)
				total_volume += R.volume
			. += "<span class='notice'>[total_volume] units of various reagents</span>"
	else
		. += "<span class='notice'>Nothing.</span>"
```